code/game/machinery/teleporter.dm

```diff
@@ -87,86 +89,70 @@
 		data["teleporterhub"] = null
 		data["calibrated"] = null
 		data["accurate"] = null
-	data["regime"] = regime_set
+	data["isGate"] = isGate
 	var/area/targetarea = get_area(target)
 	data["target"] = (!target || !targetarea) ? "None" : sanitize(targetarea.name)
 	data["calibrating"] = calibrating
 	data["locked"] = locked
 	return data
 
-/obj/machinery/computer/teleporter/Topic(href, href_list)
+/obj/machinery/computer/teleporter/tgui_act(action, params)
 	if(..())
-		return 1
-
-	if(href_list["eject"])
-		eject()
-		SSnanoui.update_uis(src)
-		return
+		return TRUE
```

Comment:
It is no longer necessary to `return TRUE` here. Just a regular `return` will do